clients/importapi/importapi.go

```diff
@@ -0,0 +1,246 @@
+type Clienter interface {
+	GetImportJob(ctx context.Context, importJobID string) (ImportJob, bool, error)
+	UpdateImportJobState(ctx context.Context, jobID string, newState string) error
+}
+
+// Client is an import api client which can be used to make requests to the API
+type Client struct {
+	client common.RCHTTPClienter
+	url    string
+}
+
+// NewAPIClient creates a new API Client with initial rchttp client (optional), given url, auth token
+func NewAPIClient(client common.RCHTTPClienter, apiURL, serviceToken string) *Client {
+	return &Client{
+		client: rchttp.ClientWithServiceToken(client, serviceToken),
+		url:    apiURL,
+	}
+}
+
+// ErrInvalidAPIResponse is returned when the api does not respond with a valid status
+type ErrInvalidAPIResponse struct {
+	actualCode int
+	uri        string
+	body       string
+}
+
+// Error should be called by the user to print out the stringified version of the error
+func (e ErrInvalidAPIResponse) Error() string {
+	return fmt.Sprintf(
+		"invalid response: %d from %s: %s, body: %s",
+		e.actualCode,
+		service,
+		e.uri,
+		e.body,
+	)
+}
+
+// Code returns the status code received from the api if an error is returned
+func (e ErrInvalidAPIResponse) Code() int {
+	return e.actualCode
+}
+
+var _ error = ErrInvalidAPIResponse{}
+
+// Healthcheck calls the healthcheck endpoint on the api and alerts the caller of any errors
+func (c *Client) Healthcheck() (string, error) {
+	ctx := context.Background()
+
+	resp, err := c.client.Get(ctx, c.url+"/healthcheck")
+	if err != nil {
+		return service, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return service, NewAPIResponse(resp, "/healthcheck")
+	}
+
+	return service, nil
+}
+
+// ImportJob comes from the Import API and links an import job to its (other) instances
+type ImportJob struct {
+	JobID string  `json:"id"`
+	Links LinkMap `json:"links,ignoreempty"`
+}
+
+// LinkMap is an array of instance links associated with am import job
+type LinkMap struct {
+	Instances []InstanceLink `json:"instances"`
+}
+
+// InstanceLink identifies an (instance or import-job) by id and url (from Import API)
+type InstanceLink struct {
+	ID   string `json:"id"`
+	Link string `json:"href"`
+}
+
+// GetImportJob asks the Import API for the details for an Import job
+func (api *Client) GetImportJob(ctx context.Context, importJobID string) (ImportJob, bool, error) {
+	var importJob ImportJob
+	path := api.url + "/jobs/" + importJobID
+
+	jsonBody, httpCode, err := api.getJson(ctx, path, 0, nil)
+	if httpCode == http.StatusNotFound {
+		return importJob, false, nil
+	}
+	logData := log.Data{
+		"path":        path,
+		"importJobID": importJobID,
+		"httpCode":    httpCode,
+		"jsonBody":    string(jsonBody),
+	}
+	var isFatal bool
+	if err == nil && httpCode != http.StatusOK {
+		if httpCode < http.StatusInternalServerError {
+			isFatal = true
+		}
+		err = errors.New("Bad response while getting import job")
+	} else {
```

Comment:
Dont think the else is required, not even sure the isFatal is actually adding anything. Where the err != nil, it will be caught by the following statement, and where the status is actually OK we probably don't want to set it to true? In either case its not being used as in the last return, so why not just directly return true in the `if err != nil`

Possbily 122 should be `if err != nil  || isFatal` to capture the correct cases?

Reply:
both routes of this `if` will ensure `err` is set and hence trigger the follow (`if err != nil {`)
*but* only one of those routes will leave `isFatal == false` (when `httpCode >= 500` server error or worse - i.e. may be worth retrying)

any other/real error sets `isFatal` to `true`

hope that clarifies

(though, it does look unclear, I admit)